astroid/brain/brain_dataclasses.py

```diff
@@ -243,15 +248,17 @@ def infer_dataclass_field_call(
     if not isinstance(node.parent, (AnnAssign, Assign)):
         raise UseInferenceDefault
     field_call = node.parent.value
```

Comment:
Isn't `node.parent.value == node`? `node` can't be `node.parent.target(s)` as it's not possible to assign something to `Call` nodes. Maybe it could make sense to substitute all `field_call` names with `node` or at least at `cast(Call, ...) here?

Reply:
Good catch! Changed!